spswarehouse/powerschool/powerschool.py

```diff
@@ -0,0 +1,437 @@
+    """
+    elem = WebDriverWait(driver, 30).until(EC.presence_of_element_located((By.NAME, element_name)))
+    elem.clear()
+    elem.send_keys(input_to_type)
+
+def powerschool_report_helper_select_visible_text_in_element_by_id(driver: WebDriver, element_id: str, text_to_select: str):
+    """
+    Waits for an element by ID and selects it by specified text.
+    """
+    elem = WebDriverWait(driver, 30).until(EC.presence_of_element_located((By.ID, element_id)))
+    select = Select(elem)
+    select.select_by_visible_text(text_to_select)
+
+def powerschool_report_helper_select_visible_text_in_element_by_name(driver: WebDriver, element_name: str, text_to_select: str):
+    """
+    Waits for an element by name and selects it by specified text.
+    """
+    elem = WebDriverWait(driver, 30).until(EC.presence_of_element_located((By.NAME, element_name)))
+    select = Select(elem)
+    select.select_by_visible_text(text_to_select)
+
+def powerschool_report_helper_click_element_by_id(driver: WebDriver, element_id: str):
+    """
+    Waits for an element by ID and clicks it.
+    """
+    elem = WebDriverWait(driver, 30).until(EC.presence_of_element_located((By.ID, element_id)))
+    elem.click()
+
+def wait_for_new_file_in_folder(folder_path, original_files, max_attempts=20000):
+    """
+    Waits until a new file shows up in a folder. Loops until that is true or max_attempts is reached.
+
+    Parameters:
+    folder_path: The folder being monitored.
+    original_files: The list of files originally in the folder, before the new one is added.
+    max_attmepts: Optional parameter that sets how many loops the function will do.
+
+    Returns:
+    n/a
+    """
+    file_added = False
+    attempts = 0
+    while True and attempts < max_attempts:
+        for root, folders, files in os.walk(folder_path):
+            # break 'for' loop if files found
+            if len(files) > len(original_files):
+                file_added = True
+                break
+            else:
+                continue
+        # break 'while' loop if files found
+        if file_added:
+            # wait for download to complete fully after it's been added - hopefully 3 seconds is enough.
+            time.sleep(3)
+            break
+        attempts +=1
+
+def rename_recent_file_in_dir(folder, append_text):
+    """
+    Gets the most recent file in a folder and appends text to its filename.
+
+    Parameters:
+    folder: The path of the folder
+    append_text: The text to appended to the filename before the extension.
+
+    Returns:
+    n/a
+    """
+
+    recent_file = get_most_recent_file_in_dir(folder)
+    recent_file = recent_file.replace('\\', '/')
+    new_file, file_ext = os.path.splitext(recent_file)
+    new_file += append_text
+    new_file += file_ext
+    os.rename(recent_file, new_file)
+
+def download_latest_report_from_report_queue_reportworks(driver: WebDriver, destination_directory_path: str = '', file_postfix: str = ''):
```

Comment:
Also seems like an internal function?

Reply:
Hmm, ok, after finishing looking at this file, maybe this isn't an internal function. This does seem like the actual usable function in the end.